### PR TITLE
fix(credentials): create credentials from teamName and not domain

### PR DESCRIPTION
**Pull request by toymachiner62**

### CODE REVIEW

File: lib/Clients/Credentials.php [hunk 1]
```diff
@@ -264,4 +266,17 @@ private function createCredentials(InternalCredentials $internalCredentials): Us
...
+     * Gets the domain from the passed in teamName
+     *
+     * @param string $teamName  The teamName to get the domain from
+     * @return string           The domain
+     */
+    private function getDomainFromTeamName(string $teamName): string
+    {
+        $pieces = explode('.', $teamName);
+        $domain = $pieces[0];
+        return $domain;
```
Comment (toymachiner62) [inline lib/Clients/Credentials.php:280]: If no period, assume it's just the team name. If there is a period, assume it's a domain name.

Are there any edge cases this misses?

Comment (toymachiner62) [reply lib/Clients/Credentials.php:280]: Wait this isn't right. I'm returning the teamName and not the domain.

File: lib/Clients/Credentials.php [hunk 2]
```diff
@@ -264,4 +266,21 @@ private function createCredentials(InternalCredentials $internalCredentials): Us
...
+    private function getDomainFromTeamName(string $teamNameOrDomain): string
+    {
+        // If the domain was passed in, return it
+        if (strpos($teamNameOrDomain, '.') === true) {
+            return $teamNameOrDomain;
+        }
+
+        // Otherwise teamName was passed in so build the domain and return it
+        return "$teamName.pushbot.com";
+    }
```
Comment (nrod80) [inline lib/Clients/Credentials.php:285]: It would be awesome to verify that the team name/domain is valid here, if this feels like a good place to do it

Comment (toymachiner62) [reply lib/Clients/Credentials.php:285]: and valid team names are any combo of letters, numbers, dashes, and underscores of any case?

Comment (nrod80) [reply lib/Clients/Credentials.php]: Pretty much, a bit of nuance though: https://github.com/catalyticlabs/pushbot-api/blob/82525eb2d19e01b615221f4cf442d66dba135860/lib/api/teams/teamModel.js#L152-L155

**Pull request merged**